errors/B0003.md

```diff
@@ -34,10 +34,8 @@ fn despawning(mut commands: Commands, entity: Option<Res<MyEntity>>) {
     }
 }
 
-fn use_entity(mut commands: Commands, entity: Option<Res<MyEntity>>) {
-    if let Some(my_entity) = entity {
-        commands.entity(my_entity.0).insert(Hello);
-    }
+fn use_entity(mut commands: Commands, my_entity: Res<MyEntity>) {
+    commands.entity(my_entity.0).insert(Hello);
```

Comment:
not sure about this change

Reply:
It's supposed to panic.

Reply:
it's not supposed to panic on the resource missing, it was supposed to panic on the missing entity

Reply:
That never happens because the resource and the entity get removed/despawn at the same time.

Reply:
That was probably changed in #9822, and should be fixed. It used to panic on the missing entity

Reply:
How should it be fixed?

Reply:
with `after` now introducing a sync point, it's going to be harder to reliably cause the panic. with two systems interacting, it would panic randomly 50% of the time depending on the order of the systems...
maybe have each system try to act on an entity despawned by the other

Reply:
I made the system no longer remove the resource, only the entity. So it now panics on the entity not existing as expected. Also modified the panic messages in the file.

Reply:
Not convinced this is properly resolved 🤔

Reply:
with your updated example, you're hitting the command creation error, so not the panic expected in that document. Maybe we could try to update this code sample in another PR?

Reply:
> you're hitting the command creation error, so not the panic expected in that document

Is it even possible to cause `[B0003]` without some weird code anymore? Yeah I agree this is probably that should be done in another PR.

Reply:
I reverted all changes in `B0003.md` because of the existance of https://github.com/bevyengine/bevy/pull/11162.

Reply:
thanks! I had in mind to update the example, but hadn't had time yet to find back this PR